Crud

hugocruz1234 opened the pull request:
Now CRUD actions work as intended, buttons correctly direct to its own pages and views, also added some validations to the model, and events are displayed as a list in a table to easier looks.

For more informations refer to  #4 
![image](https://github.com/BrightCoders-Institute/reto-administrador-de-eventos-jdg/assets/6729543/8b1bf35d-36e1-4fb7-8c97-12cd198d1b98)

![image](https://github.com/BrightCoders-Institute/reto-administrador-de-eventos-jdg/assets/6729543/48ebb787-af5b-4998-b463-2b60d11ac5eb)

![image](https://github.com/BrightCoders-Institute/reto-administrador-de-eventos-jdg/assets/6729543/dd1a34d6-b97b-48d0-884e-9aaee7fd7cfe)

![image](https://github.com/BrightCoders-Institute/reto-administrador-de-eventos-jdg/assets/6729543/bdd104a2-4b8c-47a9-a103-5f31baae234d)

![image](https://github.com/BrightCoders-Institute/reto-administrador-de-eventos-jdg/assets/6729543/1e5f6b4c-aad7-4843-b6c6-8ac4681ff7f8)

-Institute/reto-administrador-de-eventos-jdg/assets/42016840/8070805b-77a7-43b1-b739-d82a1201fb1f)

Commit:
Implement create event
hugocruz1234 cruzhugo1997@gmail.com
sr-meh hector.m.rm@hotmail.com
ballo1988 rounderjards@gmail.com
JoseLopezSolis jlopez114@ucol

Ballo1988 committed:
New function now working

co-authored-by: hugocruz1234 cruzhugo1997@gmail.com
co-authored-by: sr-meh hector.m.rm@hotmail.com
co-authored-by: JoseLopezSolis jlopez114@ucol.mx

Commit:
CRUD actions are finished
sr-meh hector.m.rm@hotmail.com
ballo1988 rounderjards@gmail.com
JoseLopezSolis jlopez114@ucol

cfcortesmx commented on app/assets/stylesheets/application.scss:
```diff
@@ -13,4 +13,20 @@
  */
- @import "bootstrap";
+@import "bootstrap";
+/* Base */
```
The 'Base' section should be placed at the top of the file, before any other imports or styles.

cfcortesmx commented on app/assets/stylesheets/application.scss:
```diff
@@ -13,4 +13,20 @@
+    font-size: 25px;
+    font-weight: bold;
+  }
+  .hover-span:hover {
```
The class name 'hover-span' is not descriptive. Consider using a more meaningful name.

cfcortesmx commented on app/assets/stylesheets/application.scss:
```diff
@@ -13,4 +13,20 @@
+  .hover-span:hover {
+    border-radius: 50%;
+  }
+  .hover-span:hover {
```
The 'cursor' property should be applied to the 'hover-span' class, not the parent 'body' element.

cfcortesmx commented on app/controllers/events_controller.rb:
```diff
@@ -1,3 +1,48 @@
 class EventsController < ApplicationController
```
The title of the pull request should be more descriptive and specific.

cfcortesmx commented on app/controllers/events_controller.rb:
```diff
@@ -1,3 +1,48 @@
-  def index; end
-end
+  def index
+    @events = Event.all
```
The `index` action should have a corresponding view to display the events.

juperameza replied:
This is important

Ballo1988 replied:
Done!

cfcortesmx commented on app/controllers/events_controller.rb:
```diff
@@ -1,3 +1,48 @@
+  end
+
+  def show
+    @event = Event.all
```
The `show` action should find a single event, not all events.

juperameza replied:
Same here

Ballo1988 replied:
Thanks, resolved!

cfcortesmx commented on app/controllers/events_controller.rb:
```diff
@@ -1,3 +1,48 @@
+  end
+
+  def new
+    @event = Event.new
```
The `new` action should create a new event instance, not find an existing event.

cfcortesmx commented on app/controllers/events_controller.rb:
```diff
@@ -1,3 +1,48 @@
+  end
+
+  def create
+    @event = Event.new(events_params)
```
The `create` action should use `event_params` instead of `events_params`.

cfcortesmx commented on app/controllers/events_controller.rb:
```diff
@@ -1,3 +1,48 @@
+    end
+  end
+
+  def edit
```
The `edit` action should find a single event, not all events.

cfcortesmx commented on app/controllers/events_controller.rb:
```diff
@@ -1,3 +1,48 @@
+  end
+
+  def update
+    @event = Event.find(params[:id])
```
The `update` action should find a single event, not all events.

cfcortesmx commented on app/controllers/events_controller.rb:
```diff
@@ -1,3 +1,48 @@
+  end
+
+  def destroy
+    @event = Event.find(params[:id])
```
The `destroy` action should find a single event, not all events.

cfcortesmx commented on app/controllers/events_controller.rb:
```diff
@@ -1,3 +1,48 @@
+    redirect_to root_path, status: :see_other
+  end
+
+  def events_params
```
The `event_params` method should be private.

juperameza replied:
This is important

Ballo1988 replied:
Already done

cfcortesmx commented on app/models/event.rb:
```diff
@@ -0,0 +1,5 @@
+class Event < ApplicationRecord
+  validates :title, presence: true
```
Consider adding a comment explaining the purpose of this validation.

cfcortesmx commented on app/models/event.rb:
```diff
@@ -0,0 +1,5 @@
+class Event < ApplicationRecord
+  validates :title, presence: true
+  validates :date, presence: true  
```
Consider adding a comment explaining the purpose of this validation.

cfcortesmx commented on app/models/event.rb:
```diff
@@ -0,0 +1,5 @@
+class Event < ApplicationRecord
+  validates :title, presence: true
+  validates :date, presence: true  
+  validates :location, presence: true 
```
Consider adding a comment explaining the purpose of this validation.

cfcortesmx commented on app/views/events/_form.html.erb:
```diff
@@ -0,0 +1,40 @@
+  <div>
+    <%= form.label :title %><br>
+    <%= form.text_field :title %>
+    <% @event.errors.full_messages_for(:title).each do |message| %>
```
Consider moving the error message display logic to a separate partial or helper method to keep the form code clean and readable.

cfcortesmx commented on app/views/events/_form.html.erb:
```diff
@@ -0,0 +1,40 @@
+  <div>
+    <%= form.label :date %><br>
+    <%= form.date_field :date %>
+    <% @event.errors.full_messages_for(:date).each do |message| %>
```
Consider moving the error message display logic to a separate partial or helper method to keep the form code clean and readable.

cfcortesmx commented on app/views/events/_form.html.erb:
```diff
@@ -0,0 +1,40 @@
+  <div>
+    <%= form.label :location %><br>
+    <%= form.text_area :location %>
+    <% @event.errors.full_messages_for(:location).each do |message| %>
```
Consider moving the error message display logic to a separate partial or helper method to keep the form code clean and readable.

cfcortesmx commented on app/views/events/index.html.erb:
```diff
@@ -1,5 +1,4 @@
-<div class="container">
-  <div class="row text-center">
+<div class="container d-flex justify-content-center">
```
Consider removing this line of code.

cfcortesmx commented on app/views/events/index.html.erb:
```diff
@@ -1,5 +1,4 @@
-<div class="container">
-  <div class="row text-center">
+<div class="container d-flex justify-content-center">
     <h1>Events Management System</h1>
```
Consider removing this line of code.

cfcortesmx commented on app/views/events/index.html.erb:
```diff
@@ -1,5 +1,4 @@
+<div class="container d-flex justify-content-center">
     <h1>Events Management System</h1>
-  </div>
 </div>
```
Consider removing this line of code.

cfcortesmx commented on app/views/events/index.html.erb:
```diff
@@ -1,5 +1,4 @@
     <h1>Events Management System</h1>
-  </div>
 </div>
+
```
Consider removing this line of code.

cfcortesmx commented on app/views/navbar/_navbar.html.erb:
```diff
@@ -0,0 +1,11 @@
+<nav class="navbar navbar-expand-lg navbar-light bg-light">
```
Consider adding a comment explaining the purpose of this file.

cfcortesmx commented on app/views/navbar/_navbar.html.erb:
```diff
@@ -0,0 +1,11 @@
+<nav class="navbar navbar-expand-lg navbar-light bg-light">
+    <div class="container">
+      <a class="navbar-brand" href="/">Brighcoders Events</a>
```
Consider adding a comment explaining the purpose of this element.

cfcortesmx commented on app/views/navbar/_navbar.html.erb:
```diff
@@ -0,0 +1,11 @@
+        <span class="navbar-toggler-icon"></span>
+      </button>
+      <div class="collapse navbar-collapse justify-content-end" id="navbarNav">
+        <a class="btn btn-primary ms-lg-3" href="/events/show">Iniciar</a>
```
The href attribute should point to the correct path for the 'Iniciar' button.

cfcortesmx commented on config/routes.rb:
```diff
@@ -1,3 +1,5 @@
 Rails.application.routes.draw do
```
The route for the events index page is missing. Please add it back.

juperameza commented on app/assets/stylesheets/application.scss:
```diff
@@ -13,4 +13,20 @@
+@import "bootstrap";
+/* Base */
+body {
+  padding: 0;
+  margin: 0;
```
Order this properties alphabetically

juperameza commented on app/controllers/events_controller.rb:
```diff
@@ -1,3 +1,48 @@
+    params.require(:event).permit(:title, :description, :date, :location, :cost)
+  end
+
+end
```
New line at the end of this file

juperameza commented on app/views/events/_form.html.erb:
```diff
@@ -0,0 +1,40 @@
+    <%= form.submit %>
+  </div>
+
+<% end %>
```
New line at the end of this file

juperameza commented on app/views/events/show.html.erb:
```diff
@@ -0,0 +1,49 @@
+      <th scope="col">Cost</th>
+      <th colspan="2" scope="col">Actions</th>
+      <a class="btn" href="/events/new">New event</a>
+    </tr>
+  </thead>
+  <tbody>
+    <% @event.each do |event| %>
+      <tr>
+        <td>
+          <%= event.title %>
+        </td>
+        <td>
+          <%= event.description %>
+        </td>
+        <td>
+          <%= event.date %>
+        </td>
+        <td>
+          <%= event.location %>
+        </td>
+        <td>
+          $<%= event.cost %>
+        </td>
+        <td>
+          <%= link_to "Edit", edit_event_path(event), class: "material-symbols-outlined hover-span" %>
+        </td>
+        <td>
+          <li>
+            <%= link_to "Delete", event_path(event), method: :delete, data: {
+              turbo_method: :delete,
+              turbo_confirm: "Are you sure?"
+            }, class: "material-symbols-outlined hover-span" %>
+          </li>
+          
+        </td>
+      </tr>
+    <% end %>
+  </tbody>
+</table>
+</div>
```
This should be the index file

juperameza commented on app/controllers/events_controller.rb:
```diff
@@ -1,3 +1,48 @@
+    end
+  end
+
+  def edit
+    @event = Event.find(params[:id])
+  end
+
+  def update
+    @event = Event.find(params[:id])
+
+    if @event.update(events_params)
+      redirect_to @event
+    else
+      render :edit, status: :unprocessable_entity
+    end
+  end
+
+  def destroy
+    @event = Event.find(params[:id])
```
To keep this [DRY](https://thevaluable.dev/dry-principle-cost-benefit-example/) consider adding a set_event method. Something like this ` def set_post
      @post = Post.find(params[:id])
    end` and call a before action and the top of the controller

Ballo1988 replied:
Done

juperameza reviewed (changes requested):
Looking at the branches graph you it seems you did a branch from main and then did a branch from that branch. This will lead you to a lot of conflicts, you should avoid doing this, all new branches should come from main. Take a look at this [POST](https://www.atlassian.com/git/tutorials/comparing-workflows/feature-branch-workflow)

sr-meh committed:
merge main into crud branch
co-authored-by: hugocruz1234 cruzhugo1997@gmail.com
co-authored-by: JoseLopezSolis jlopez114@ucol.mx
co-authored-by : ballo1988 rounderjards@gmail.com

sr-meh committed:
The code was modified based on the suggestions received, in addition to adding styles to the views based on boostrap.
co-authored-by: hugocruz1234 cruzhugo1997@gmail.com
co-authored-by: JoseLopezSolis jlopez114@ucol.mx
co-authored-by : ballo1988 rounderjards@gmail.com

sr-meh committed:
update the styles
co-authored-by: hugocruz1234 cruzhugo1997@gmail.com
co-authored-by: JoseLopezSolis jlopez114@ucol.mx
co-authored-by : ballo1988 rounderjards@gmail.com

juperameza commented on config/routes.rb:
```diff
@@ -1,5 +1,7 @@
 Rails.application.routes.draw do
   root "events#index"
 
+  get '/events/show', to: 'events#show', as: 'show_event'
```
This isnt needed you have your resources

juperameza replied:
Explained in the call. They needed this due to other errors. They needed another controller for the index but the requirements where amnbigous

Commit:
Modified controller
sr-meh hector.m.rm@hotmail.com
ballo1988 rounderjards@gmail.com
JoseLopezSolis jlopez114@ucol

juperameza reviewed (approved):
LGTM

hugocruz1234 committed:
Merge branch 'main' into crud

Ballo1988 reviewed (approved):
Well done guys!
